Src/AutoFixtureUnitTest/FreezeOnMatchCustomizationTest.cs

```diff
@@ -77,6 +90,63 @@ public void InitializeWithMatcherShouldSetCorrespondingProperty()
             Assert.Equal(matcher, sut.Matcher);
         }
 
+
+        [Fact]
+        public void InitializeWithSingleNullRequestArgumentShouldThrowArgumentNullException()
+        {
+            Assert.Throws<ArgumentNullException>(() =>
+                new FreezeOnMatchCustomization((object)null));
+        }
+
+        [Fact]
+        public void InitializeWithNullRequestShouldThrowArgumentNullException()
+        {
+            var ex = Assert.Throws<ArgumentNullException>(() =>
+                new FreezeOnMatchCustomization((object)null, new FalseRequestSpecification()));
+            Assert.Equal("request", ex.ParamName);
+        }
+
+        [Fact]
+        public void InitializeWithRequestAndNullMatcherShouldThrowArgumentNullException()
+        {
+            var ex = Assert.Throws<ArgumentNullException>(() =>
+                  new FreezeOnMatchCustomization(new object(), null));
+            Assert.Equal("matcher", ex.ParamName);
+        }
+
+        [Fact]
+        public void InitializeWithSingleRequestShouldSetCorrespondingProperty()
+        {
+            var request = new object();
+            var sut = new FreezeOnMatchCustomization(request);
+            Assert.Same(request, sut.Request);
+        }
+
+        [Fact]
+        public void InitializeWithObjectRequestShouldSetTrueRequestSpecificationMatcher()
```

Comment:
Are you talking about [Structural Inspection anti-pattern](http://enterprisecraftsmanship.com/2016/07/21/unit-testing-anti-patterns-structural-inspection/)? Yep, that is a structural test. I suppose it was added mechanically following the test-first practice.

Despite the code change _is_ covered by the other tests in [the same](https://github.com/AutoFixture/AutoFixture/pull/697/commits/4cd5ee87615aa62e698808671cee6748a863caa7) commit, I do not see any issues in keeping this 'structural' test. It's tiny and it may serve the code documentation function much better than the [`CustomizeWithNotEqualRequestsShouldNotFreezeSpecimen`](https://github.com/AutoFixture/AutoFixture/pull/697/files#diff-3b20e30bf8f7c8e5020d5f1e40f2907bR234) test which checks that the behavior is correct.